metrics.py

```diff
@@ -332,24 +332,20 @@ def get_columns_to_report(dataset_columns):
         'literature'
     ]
 
-
-def detect_spark_memory_limit():
-    """Spark does not automatically use all available memory on a machine. When working on large datasets, this may
-    cause Java heap space errors, even though there is plenty of RAM available. To fix this, we detect the total amount
-    of physical memory and allow Spark to use (almost) all of it."""
-    mem_bytes = os.sysconf('SC_PAGE_SIZE') * os.sysconf('SC_PHYS_PAGES')
-    mem_gib = math.floor(mem_bytes/(1024**3) * 0.90)
-    return mem_gib
```

Comment:
I would suggest not removing this function, but rather making it less fragile on OSX. For example, you could check the platform (`if sys.platform == 'linux'`) and only do the computations in that case, and return some sensible hardcoded value for a Mac.

This is especially important for this module, because it crunches a lot of data in a lot of different ways. It could be run very fast on a high memory machine, but if we hardcode 15 GB of memory, the capabilities of such a machine would be heavily underutilised.

Reply:
I've reimplemented the function by using [psutil](https://www.geeksforgeeks.org/psutil-module-in-python/), which has cross-platform support. 
Fixed in d92bed6.